core-plugin/src/main/groovy/com/github/jrubygradle/api/gems/GemVersion.groovy

```diff
@@ -0,0 +1,573 @@
+            String exact = getVersionFromRequirement(singleRequirement, EQUAL)
+            new GemVersion(
+                INCLUSIVE,
+                exact,
+                exact,
+                INCLUSIVE
+            )
+        } else if (singleRequirement.matches(NOT_EQUAL)) {
+            log.info("'${singleRequirement}' is supported by Ivy.")
+            NO_VERSION
+        } else if (singleRequirement.matches(LESS_EQUAL)) {
+            new GemVersion(
+                OPEN_ENDED,
+                null,
+                getVersionFromRequirement(singleRequirement, LESS_EQUAL),
+                INCLUSIVE
+            )
+        } else if (singleRequirement.matches(LESS)) {
+            new GemVersion(
+                OPEN_ENDED,
+                null,
+                getVersionFromRequirement(singleRequirement, LESS),
+                EXCLUSIVE
+            )
+        } else if (singleRequirement.matches(TWIDDLE_WAKKA)) {
+            parseTwiddleWakka(singleRequirement)
+        } else {
+            throw new GemVersionException("'${singleRequirement}' does not look like a GEM version requirement")
+        }
+    }
+
+    /** Is the low version specification inclusive?
+     *
+     * @return {@code true} if inclusive.
+     *
+     * @since 2.0
+     */
+    boolean isLowInclusive() {
+        lowBoundary == INCLUSIVE
+    }
+
+    /** Is the high version specification inclusive?
+     *
+     * @return {@code true} if inclusive.
+     *
+     * @since 2.0
+     */
+    boolean isHighInclusive() {
+        highBoundary == INCLUSIVE
+    }
+
+    /** Is the high version unspecified?
+     *
+     * @return {@code true} if the high version is unspecified in the original GEM specification.
+     *
+     * @since 2.0
+     */
+    boolean isHighOpenEnded() {
+        highBoundary == Boundary.OPEN_ENDED
+    }
+
+    /**
+     * since GemVersion is version range with lower bound and upper bound
+     * this method just calculates the intersection of this version range
+     * with the given other version range. it also honors whether the boundary
+     * itself is included or excluded by the respective ranges.
+     *
+     * @param The other version range to be intersected with this version range
+     * @return GemVersion the intersected version range
+     */
+    GemVersion intersect(String otherVersion) {
+        intersect(gemVersionFromGradleIvyRequirement(otherVersion))
+    }
+
+    /**
+     * since GemVersion is version range with lower bound and upper bound
+     * this method just calculates the intersection of this version range
+     * with the given other version range. it also honors whether the boundary
+     * itself is included or excluded by the respective ranges.
+     *
+     * @param The other version range to be intersected with this version range
+     * @return GemVersion the intersected version range
+     *
+     * @since 2.0
+     */
+    GemVersion intersect(GemVersion other) {
```

Comment:
The logic is mostly the same. The real difference is the use of the enum and to accomodate open ranges as well.